gamla/io_utils.py

```diff
@@ -172,10 +172,14 @@ def throttle(limit, f):
 
     >>> throttled_get_async = throttle(3, get_async)
     """
-    semaphore = asyncio.Semaphore(limit)
+    semaphore = None
 
     @functools.wraps(f)
     async def wrapped(*args, **kwargs):
+        nonlocal semaphore
+        # Initialize semaphore lazily to allow applying `throttle` when an event loop does not yet exist.
```

Comment:
```suggestion
        # This must be in the inner function so that we avoid creating an event loop before the user has, causing the code to run with two different event loops.
```